modules/adagioAnalyticsAdapter.js

```diff
@@ -378,10 +386,34 @@ function handlerAdRender(event, isSuccess) {
   sendNewBeacon(auctionId, adUnitCode);
 };
 
+function handleBidTimeout(event) {
+  const { adUnitCode } = event.bid;
+  let auctionId = getTargetedAuctionId(event.bid);
+  cache.updateAuction(auctionId, adUnitCode, {
+    bdrs_timeout: true
+  });
+};
+
 /**
  * END HANDLERS
  */
 
+function subscribeToGamSlots() {
+  window.googletag.pubads().addEventListener('slotRenderEnded', event => {
+    const auction = cache.getAuctionByAdunit(event.slot.getAdUnitPath(), event.slot.getSlotElementId());
+    if (!auction) {
+      const slotName = `${event.slot.getAdUnitPath()} - ${event.slot.getSlotElementId()}`;
+      logWarn('Could not find configured ad unit matching GAM render of slot: ' + slotName);
+      return;
+    }
+
+    cache.updateAuction(auction.auct_id, auction.adu_code, {
+      adsrv: 'gam',
+      adsrv_empty: event.isEmpty
+    });
+  });
+}
+
```

Comment:
Not urgent: It would be nice to create this function in a more generic way in `library/gptUtils/gptUtils.js`. By the way we'll potentially leverage "code duplication" alert.